test/suite/hook_tiered.py

```diff
@@ -300,89 +300,36 @@ def uses(self, use_list):
-                "test_import04.test_table_import",
-                "test_import09.test_import_table_repair",
-                "test_import09.test_import_table_repair",
-                "test_import11.test_file_import",
-                "test_import11.test_file_import",
-                "test_join03.test_join",
-                "test_join07.test_join_string",
-                "test_jsondump02.test_json_all_bytes",
-                "test_metadata_cursor02.test_missing",
-                "test_prepare02.test_prepare_session_operations",
-                "test_prepare_hs03.test_prepare_hs",
-                "test_prepare_hs03.test_prepare_hs",
-                "test_rename.test_rename",
-                "test_rollback_to_stable09.test_rollback_to_stable",
-                "test_rollback_to_stable28.test_update_restore_evict_recovery",
-                "test_rollback_to_stable34.test_rollback_to_stable",
-                "test_rollback_to_stable35.test_rollback_to_stable",
-                "test_rollback_to_stable36.test_rollback_to_stable",
-                "test_sweep03.test_disable_idle_timeout_drop",
-                "test_sweep03.test_disable_idle_timeout_drop_force",
-                "test_txn22.test_corrupt_meta",
-                "test_verbose01.test_verbose_single",
-                "test_verbose02.test_verbose_single",
-                "test_verify2.test_verify_ckpt",
-                ]
-
-        for item in skip:
-            if item in str(test):
-                return True
-        return False
-
-    # Remove tests that won't work on tiered cursors
-    def filter_tests(self, tests):
-        new_tests = unittest.TestSuite()
-        new_tests.addTests([t for t in tests if not self.skip_test(t)])
-        return new_tests
+    # Determine if a test should be skipped. 
+    # If it should skip, return the reason as a string. If it shouldn't, return None
+    # Some features aren't supported with tiered storage currently. If they exist in 
+    # the test name (or scenario name) skip the test.
+    def skip_reason(self, test):
+        skip_categories = [
+            ("backup",               "Can't backup a tiered table"),
+            ("inmem",                "In memory tests don't make sense with tiered storage"),
+            ("lsm",                  "If the test name tells us it uses lsm ignore it"),
+            ("modify_smoke_recover", "Copying WT dir doesn't copy the bucket directory"),
+            ("test_salvage",         "Salvage tests directly name files ending in '.wt'"),
+            ("test_config_json",     "create replacement can't handle a json config string"),
+            ("test_cursor_big",      "Cursor caching verified with stats"),
+            ("tiered",               "Tiered tests already do tiering."),
+            ("test_verify",          "Verify not supported on tiered tables (yet)")
+        ]
+
+        for (skip_string, skip_reason) in skip_categories:
+            if skip_string in str(test):
```

Comment:
Could we have multiple matches? I am afraid we would match and return an unexpected `skip_reason`. I wonder if we could make sure this never happens.

Reply:
We probably don't need to show all the reasons why, knowing that it is skipped is the most important part. As long as we are okay with multiple matches, I am fine with the way it is, thanks for double checking!